server/mqtt.go

```diff
@@ -1523,51 +1528,108 @@ func (jsa *mqttJSA) prefixDomain(subject string) string {
 	return subject
 }
 
-func (jsa *mqttJSA) newRequestEx(kind, subject, cidHash string, hdr int, msg []byte, timeout time.Duration) (interface{}, error) {
-	var sb strings.Builder
-	// Either we use nuid.Next() which uses a global lock, or our own nuid object, but
-	// then it needs to be "write" protected. This approach will reduce across account
-	// contention since we won't use the global nuid's lock.
-	jsa.mu.Lock()
-	uid := jsa.nuid.Next()
-	sb.WriteString(jsa.rplyr)
-	jsa.mu.Unlock()
-
-	sb.WriteString(kind)
-	sb.WriteByte(btsep)
-	if cidHash != _EMPTY_ {
-		sb.WriteString(cidHash)
-		sb.WriteByte(btsep)
+func (jsa *mqttJSA) newRequestEx(kind, subject, cidHash string, hdr int, msg []byte, timeout time.Duration) (any, error) {
+	responses, err := jsa.newRequestExMulti(kind, subject, cidHash, []int{hdr}, [][]byte{msg}, timeout)
+	if err != nil {
+		return nil, err
+	}
+	if len(responses) != 1 {
+		return nil, fmt.Errorf("unreachable: invalid number of responses (%d)", len(responses))
+	}
+	r := responses[0]
+	type errorer interface{ ToError() error }
+	if e, ok := r.value.(errorer); ok {
+		if err := e.ToError(); err != nil {
+			return nil, err
+		}
 	}
-	sb.WriteString(uid)
-	reply := sb.String()
+	return r.value, nil
+}
 
-	ch := make(chan interface{}, 1)
-	jsa.replies.Store(reply, ch)
+// newRequestExMulti sends multiple messages on the same subject and waits for
+// all responses. Unless it times out, it returns the same number of responses
+// in the same order as msgs parameter.
+func (jsa *mqttJSA) newRequestExMulti(kind, subject, cidHash string, hdrs []int, msgs [][]byte, timeout time.Duration) ([]*mqttJSAResponse, error) {
+	if len(msgs) == 0 {
```

Comment:
If anything, then I would consider that an error, so return `nil, fmt.Errorf("...")`. But those checks that you have added are more like a dev error, not runtime, so either should be a panic, or simply removed.

Reply:
Removed